Cron/DeleteBackups.php

```diff
@@ -0,0 +1,206 @@
+{
+    /**
+     * @var DirectoryList
+     */
+    private DirectoryList $directoryList;
+
+    /**
+     * @var DateTime
+     */
+    private DateTime $dateTime;
+
+    /**
+     * @var CronConfig
+     */
+    private CronConfig $cronConfig;
+
+    /**
+     * @var LoggerInterface
+     */
+    private LoggerInterface $logger;
+
+    /**
+     * @param DirectoryList $directoryList
+     * @param DateTime $dateTime
+     * @param CronConfig $cronConfig
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        DirectoryList $directoryList,
+        DateTime $dateTime,
+        CronConfig $cronConfig,
+        LoggerInterface $logger
+    )
+    {
+        $this->directoryList = $directoryList;
+        $this->dateTime = $dateTime;
+        $this->cronConfig = $cronConfig;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return void
+     * @throws FileSystemException
+     */
+    public function execute()
+    {
+        if ($this->cronConfig->isCronEnabled()) {
+
+            $now = $this->dateTime->gmtTimestamp();
+            $varPath = $this->directoryList->getPath(DirectoryList::VAR_DIR);
+            $cmsPath = $varPath . DS . CronConfig::CMS_DIR;
+            $directoryIterator = new \RecursiveDirectoryIterator($cmsPath, \FilesystemIterator::SKIP_DOTS);
+
+            if ($this->cronConfig->getMethodType() === CronConfig::PERIOD) {
+                $filesByPeriods = $this->findAndSortFilesByPeriods($directoryIterator, $now);
+
+                if (count($filesByPeriods)) {
+                    $deletedFiles = [];
+
+                    foreach ($filesByPeriods as $filesByPeriod) {
+                        $deletedFiles[] = $this->deleteFilesByPeriod($filesByPeriod['files'], $filesByPeriod['period']);
+                    }
+
+                    if($this->cronConfig->isLogsEnabled()) {
+                        $this->logger->debug(
+                            'CMS Content. Delete files by cron:',
+                            ['method' => CronConfig::PERIOD, 'files' => $deletedFiles]
+                        );
+                    }
+                }
+            } else if ($this->cronConfig->getMethodType() === CronConfig::OLDER_THAN) {
+                $filesOlderThan = $this->findFilesOlderThan($directoryIterator, $now);
+
+                if (count($filesOlderThan)) {
+                    $this->deleteFilesOlderThan($filesOlderThan);
+
+                    if($this->cronConfig->isLogsEnabled()) {
+                        $this->logger->debug(
+                            'CMS Content. Delete files by cron:',
+                            ['method' => CronConfig::OLDER_THAN, 'files' => $filesOlderThan]
+                        );
+                    }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * @param $directoryIterator
+     * @param $now
+     * @return array
+     */
+    private function findAndSortFilesByPeriods($directoryIterator, $now): array
```

Comment:
In case used function return type then params in function must be typed too:
findAndSortFilesByPeriods(RecursiveDirectoryIterator $directoryIterator, string $now): array